src/server/subscriptions.ts

```diff
@@ -0,0 +1,49 @@
+import * as uuidv4 from "uuid/v4";
+import { EventEmitter } from "events";
+import { augurEmitter } from "../events";
+import { ErrorCallback } from "../types";
+
+export class Subscriptions extends EventEmitter {
+  public subscribe(eventName: string, params: any, publish: (data: {}) => void): string {
+    switch (eventName) {
+      case "MarketCreated":
+        return this.subscribeToEvent(eventName, params, publish);
+      default:
+        throw new Error(`Event ${eventName} not available for subscription`);
+    }
+  }
+
+  public unsubscribe(subscription: string): void  {
+    this.emit(`unsubscribe:${subscription}`);
+  }
+
+  public removeAllListeners(eventName?: string | symbol | undefined): this {
```

Comment:
Is `this` the correct/intended return type for `removeAllListeners`?

Reply:
Also, when will the `eventName` parameter be included vs left out for this function?  If this simply removes all listeners, is the event name param needed at all?

Reply:
This is the protoype for EventEmitter#removeAllListeners, and yes the return type is `this` -- it works out to be the current class type because in JS a class is a function, and the function definitions are within that context. Weird, yes, but blame it on the prototype...

removeAllListeners can remove all listeners on the Emitter or all listeners for a specific key. This is because you can bind multiple listeners to one event for an event emitter.